bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v3/product/ProductImpl.java

```diff
@@ -37,36 +41,54 @@
 import com.adobe.cq.commerce.magento.graphql.ConfigurableProductOptionsValues;
 import com.adobe.cq.commerce.magento.graphql.ConfigurableVariant;
 
-@Model(adaptables = SlingHttpServletRequest.class, adapters = Product.class, resourceType = ProductImpl.RESOURCE_TYPE)
+@Model(
+    adaptables = SlingHttpServletRequest.class,
+    adapters = Product.class,
+    resourceType = ProductImpl.RESOURCE_TYPE)
 public class ProductImpl extends com.adobe.cq.commerce.core.components.internal.models.v2.product.ProductImpl
     implements Product {
 
     public static final String RESOURCE_TYPE = "core/cif/components/commerce/product/v3/product";
 
-    protected static final Map<String, String> SECTIONS_MAP = new HashMap<String, String>() {
+    protected static final String PN_VISIBLE_SECTIONS = "visibleSections";
+
+    protected static final Map<String, String> SECTIONS_MAP = Collections.unmodifiableMap(new HashMap<String, String>() {
         {
-            put(TITLE_SECTION, "showTitle");
-            put(PRICE_SECTION, "showPrice");
-            put(SKU_SECTION, "showSku");
-            put(IMAGE_SECTION, "showImage");
-            put(OPTIONS_SECTION, "showOptions");
-            put(QUANTITY_SECTION, "showQuantity");
-            put(ACTIONS_SECTION, "showActions");
-            put(DESCRIPTION_SECTION, "showDescription");
-            put(DETAILS_SECTION, "showDetails");
+            put("title", TITLE_SECTION);
+            put("price", PRICE_SECTION);
+            put("sku", SKU_SECTION);
+            put("images", IMAGE_SECTION);
+            put("options", OPTIONS_SECTION);
+            put("quantity", QUANTITY_SECTION);
+            put("actions", ACTIONS_SECTION);
+            put("description", DESCRIPTION_SECTION);
+            put("details", DETAILS_SECTION);
         }
-    };
+    });
+
+    private static final String[] VISIBLE_SECTIONS_DEFAULT = SECTIONS_MAP.keySet().toArray(new String[0]);
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String[] visibleSections;
+
+    private Set<String> visibleSectionsSet;
 
     @PostConstruct
     protected void initModel() {
         super.initModel();
+
         if (productRetriever != null) {
             productRetriever.extendProductQueryWith(p -> p.onConfigurableProduct(cp -> cp
                 .configurableOptions(o -> o
                     .values(v -> v.uid()))
                 .variants(v -> v
                     .attributes(a -> a.uid()))));
         }
+
+        if (visibleSections == null || visibleSections.length == 0) {
```

Comment:
When using the edit and design dialog together to define a setting, there are usually two aspects you want to cover:

- In the design dialog:
    - Define which options are available for the author in the edit dialog.
    - Define the default setting.
- In the edit dialog:
    - Customize component if you want to deviate from the default and choose between allowed settings.

This pattern is used in the WCM title component: https://github.com/adobe/aem-core-wcm-components/tree/main/content/src/content/jcr_root/apps/core/wcm/components/title/v3/title. I recommend verifying this with our PM.